src/acl/external/session/ext_session_acl.cc

```diff
@@ -137,7 +137,13 @@ static void init_db(void)
         }
     }
 #elif USE_TRIVIALDB
-    db = tdb_open(db_path, 0, TDB_CLEAR_IF_FIRST, O_CREAT|O_DSYNC, 0666);
+#if _SQUID_FREEBSD_
+    // FreeBSD lacks O_DSYNC, O_SYNC is closest to correct behaviour
+    const auto openFlags = O_CREAT|O_SYNC;
+#else
+    const auto openFlags = O_CREAT|O_DSYNC;
+#endif
+    db = tdb_open(db_path, 0, TDB_CLEAR_IF_FIRST, openFlags, 0666);
```

Comment:
We do not have to duplicate the `O_CREAT` part of the opening flags logic, right?

```suggestion
    const auto syncFlags = O_SYNC;
#else
    const auto syncFlags = O_DSYNC;
#endif
    db = tdb_open(db_path, 0, TDB_CLEAR_IF_FIRST, O_CREAT|syncFlags, 0666);
```

Reply:
We could, but AFAICT they way I did it allows the compiler to more easily deduce whether the parameter is constexpr.

Reply:
> they way I did it allows the compiler to more easily deduce whether the parameter is constexpr.

Your call, but I bet modern compilers will optimize both versions the same way: https://godbolt.org/z/2YJqg5

> One similar alternative would be a local #define for O_DSYNC instead of the local constant.

That sounds like the best option to me!

And just in case it was not obvious from the lack of a negative review with a change request: I do _not_ insist on any of these changes.

Reply:
Okay, Done.

Reply:
Thank you. Approving in anticipation of the double-tdb_open() typo fix.